Makefile

```diff
@@ -168,9 +171,14 @@ njs-fmt: ## Run prettier against the njs httpmatches module
 vet: ## Run go vet against code
 	go vet ./...
 
+.PHONY: check-golangci-lint
+check-golangci-lint:
+	@golangci-lint --version || (code=$$?; printf "\033[0;31mError\033[0m: there was a problem with golangci-lint. Follow the docs to install it https://golangci-lint.run/welcome/install/\n"; exit $$code)
+	@golangci-lint --version | grep -q $(GOLANGCI_LINT_VERSION) || (printf "\033[0;33mWarning\033[0m: your golangci-lint version is different from the one specified in .pre-commit-config.yaml. The recommended version is $(GOLANGCI_LINT_VERSION)\n")
+
 .PHONY: lint
-lint: ## Run golangci-lint against code
-	docker run --pull always --rm -v $(CURDIR):/nginx-gateway-fabric -w /nginx-gateway-fabric -v $(shell go env GOCACHE):/cache/go -e GOCACHE=/cache/go -e GOLANGCI_LINT_CACHE=/cache/go -v $(shell go env GOPATH)/pkg:/go/pkg golangci/golangci-lint:latest golangci-lint --color always run
+lint: check-golangci-lint ## Run golangci-lint against code
+	golangci-lint run
```

Comment:
Why not continue to use the docker image so that devs don't have to install/update?

Reply:
I couldn't find a way to make the command work from the `tests` directory...and it's also always been very slow.

Another approach I've seen in other projects is that they download all the tools locally automatically if they're not present in the system...but I'm not sure how I feel about that...